src/main/java/org/gridsuite/cases/importer/job/CaseImportServiceRequester.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.cases.importer.job;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+public class CaseImportServiceRequester {
+
+    private static final String API_VERSION = "v1";
+
+    private static String serviceUrl;
+
+    private final HttpClient httpClient;
+
+    public CaseImportServiceRequester(String serviceUrl) {
+        this.serviceUrl = serviceUrl;
+        httpClient = HttpClient.newHttpClient();
+    }
+
+    public boolean importCase(TransferableFile caseFile) throws IOException, InterruptedException {
+        Map<Object, Object> data = new LinkedHashMap<>();
+        data.put("file", caseFile);
+
+        // Random 256 length string is used as multipart boundary
+        String boundary = new BigInteger(256, new Random()).toString();
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(serviceUrl + "/case-server/" + API_VERSION + "/cases/public"))
+                .timeout(Duration.ofMinutes(2))
+                .header("Content-Type", "multipart/form-data;boundary=" + boundary)
+                .POST(ofMimeMultipartData(data, boundary))
+                .build();
+
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        System.out.println("Case server response status: " + response.statusCode());
+        System.out.println(response.body());
+        return (response.statusCode() == 200);
+    }
+
+    private HttpRequest.BodyPublisher ofMimeMultipartData(Map<Object, Object> data,
```

Comment:
This is quite ugly code. Is there is way to rely of third party lib for that? Is it really needed?

Reply:
It seems that HttpClient of Java 11 does not provide support for Multipart body in POST requests. It has to be implemented specifically. Apache HttpClient might have this support already provided, but do we want to add dependency to Apache HttpComponents library ?